### PR TITLE
Initial implementation of GameDetailedView

### PR DESCRIPTION
**Pull request by gnomex909**
Initial implementation, without AbilityList and GameBox. But there are already many things to comment and work on I think. Also there is an error with refreshing I sent you on Discord

### CODE REVIEW

File: frontend/components/GameBox.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	font-size: 12px;
+	margin-left: 20px;
+	display: list-item;
+	list-style: circle;
+`;
+
+export function GameBox() {
+	return (
+		<Box>
+			<BoxArt src="https://steamuserimages-a.akamaihd.net/ugc/90469956945157107/05A816D9D18F48CEE9FE3E0AA8706B0BF81B669E/" />
```
Comment (EvidentlyCube) [inline frontend/components/GameBox.tsx]: I understand it's all temporary here?

File: frontend/views/App.tsx [hunk 2]
```diff
@@ -28,6 +29,7 @@ export class App extends React.Component<AppProps> {
 								<Route path="/games" render={() => <GamesView store={this.props.store} />} />
 								<Route path="/abilities" />
 								<Route path="/config" render={() => <ConfigView store={this.props.store} />} />
+								<Route path="/game/:chosenGame?" component={GameDetailedView} />
```
Comment (EvidentlyCube) [inline frontend/views/App.tsx]: Just go with `gameId` instead of `chosenGame`

File: frontend/views/GameDetailedView.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	chosenGame: Game;
+}
+
+export class GameDetailedView extends React.Component<GameDetailevViewProps, GameDetailedViewState> {
+	chosenSeries: GameSeries;
+	chosenImage: Image;
+	constructor(props: GameDetailevViewProps) {
+		super(props);
+		this.state = {
+			chosenGame:
```
Comment (EvidentlyCube) [inline frontend/views/GameDetailedView.tsx]: What I'd do instead is make it so that if there is no game matching for the ID the user is immediately rerouted to game-list page. This will save you from having to add the clunky fallback.

This might have more info: https://www.reddit.com/r/reactjs/comments/7laeel/redirect_before_componentdidmount/

File: frontend/views/GameDetailedView.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+				props.games.get(props.chosenGameId) ||
+				new Game({ title: 'Undefined', id: -1, seriesId: -1, imageId: -1, description: 'undefined', analysis: 'undefined' }),
+		};
+		this.chosenSeries =
+			props.gameSeries.get(this.state.chosenGame.seriesId) ||
+			new GameSeries({ id: -1, name: 'undefined', description: 'undefined', wikiUrl: 'undefined' });
```
Comment (EvidentlyCube) [inline frontend/views/GameDetailedView.tsx]: There is a hard-constraint in the backend that a game MUST have a valid series ID provided. Just set the value (You can type `!` after `get()` to tell TS that you are absolutely sure it can't be undefined. Then on a separate line do a null check and throw an exception if somehow series were not found.

File: frontend/views/GameDetailedView.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+		this.chosenImage = props.images.get(this.state.chosenGame.imageId) || new Image({ id: -1, name: 'undefined', fileUrl: 'undefined' });
```
Comment (EvidentlyCube) [inline frontend/views/GameDetailedView.tsx]: Instead of creating a `new Image()` here it makes more sense to have a generic "Missing image" instance you can use anywhere. You can store it as a static member in `Image` class for now (or export a const with the default instance. Just create some temporary gray image with a question mark in the middle or something).

File: frontend/components/GameBox.tsx (hunk 1 above)
Comment (gnomex909) [reply frontend/components/GameBox.tsx]: Yap, Gamebox is placeholder ATM

File: frontend/storage/models/Image.ts [hunk 6]
```diff
@@ -1,4 +1,5 @@
 export type ImageProps = Required<Image>;
+import UndefinedCover from '../../assets/cover_undefined.jpg';
```
Comment (EvidentlyCube) [inline frontend/storage/models/Image.ts]: Go with `UndefinedCoverPath` (since it actually is a path)

Edit: Actually go with `DefaultImagePath`, more below.

File: frontend/storage/models/Image.ts [hunk 7]
```diff
@@ -12,3 +13,5 @@ export class Image {
 		this.fileUrl = props.fileUrl;
 	}
 }
+
+export const undefinedCover = new Image({ id: -1, name: 'undefined cover', fileUrl: UndefinedCover });
```
Comment (EvidentlyCube) [inline frontend/storage/models/Image.ts]: My personal preference is to use PascalCase for exported constants but a quick google tells me it's not a common way of doing things. 

Anyhow `defaultImage` would be a better fit in this case, because the goal is to use this image everywhere an image is missing.

File: frontend/views/App.tsx [hunk 8]
```diff
@@ -28,6 +29,7 @@ export class App extends React.Component<AppProps> {
 								<Route path="/games" render={() => <GamesView store={this.props.store} />} />
 								<Route path="/abilities" />
 								<Route path="/config" render={() => <ConfigView store={this.props.store} />} />
+								<Route path="/game/:gameId?" component={GameDetailedView} />
```
Comment (EvidentlyCube) [inline frontend/views/App.tsx]: Does the question mark signify `gameId` param is optional? Why did you add it?

File: frontend/views/GameDetailedView.tsx [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+			};
+			this.chosenSeries = props.gameSeries.get(this.state.chosenGame.seriesId)!;
+			if (!this.chosenSeries) {
+				throw new ReferenceError(
+					`Game with id: ${props.chosenGameId} have a missing game series associated with it. Game Series Id:${this.state.chosenGame.seriesId}`
+				);
+			}
+			this.chosenImage = props.images.get(this.state.chosenGame.imageId) || undefinedCover;
+		} else {
+			this.gameFound = false;
```
Comment (EvidentlyCube) [inline frontend/views/GameDetailedView.tsx]: There must be a better way to do this but leave it as-is for now.

Comment (EvidentlyCube) [reply frontend/views/GameDetailedView.tsx]: Okay, I don't see a better way. This is crap and I hate this, so here's a proposition:

 1. For each view have a separate `<name>Route` component, whose only job is to check preconditions and render the actual view.
 2. This would mean move the store logic to the `Route` components, and make the `View` components get the final data as props.
 3. Thus Views will become simpler, and will always work from the assumption that the data they have is valid.

It's probably best to do this after this PR is merged, so keep that in mind as your next step.
